src/extras/inputs/input.js

```diff
@@ -0,0 +1,84 @@
+class InputDelta {
+    /**
+     * @type {number[]}
+     */
+    _instance;
+
+    /**
+     * @param {number} dim - The dimension.
+     */
+    constructor(dim = 1) {
+        this._instance = new Array(dim).fill(0);
+    }
+
+    get value() {
+        return this._instance;
+    }
+
+    /**
+     * @param {number[]} offsets - The offsets.
+     */
+    add(offsets) {
+        for (let i = 0; i < this._instance.length; i++) {
+            this._instance[i] += offsets[i] || 0;
+        }
+    }
+
+    flush() {
+        this._instance.fill(0);
+    }
+}
+
+class InputDevice {
```

Comment:
`GamepadInputDevice` subclasses this and appears in the docs so ideally, this would be there too.